vernac/locality.ml

```diff
@@ -6,22 +6,12 @@
 (*         *       GNU Lesser General Public License Version 2.1        *)
 (************************************************************************)
 
-open Pp
-
 (** * Managing locality *)
 
 let local_of_bool = function
   | true -> Decl_kinds.Local
   | false -> Decl_kinds.Global
 
-let check_locality locality_flag =
```

Comment:
Wait! Where is this function gone?

Reply:
@ejgallego IIUC, there is a bit of a design flaw in this PR: it seems that vernac command not supporting locality cannot use `VERNAC COMMAND FUNCTIONAL EXTEND`, or they will have to implement themselves the logic that prints an error message if a `Local` prefix is provided. Is that correct?

Reply:
In fact, it is worse than that. Try for example:
```
Require Import FunInd.
Local Function f (n : nat) := n.
```

Previously, it was failing (admittedly, in a bit of a weird way) because `Function` does not support the `Local` prefix. Now it silently ignores `Local`, which is a regression.

Reply:
And FTR, nothing is specific to `Function` here, it will be the same for all plugin-defined vernacular commands that don't support locality.

Reply:
That doesn't seem wrong to me: in an attribute-based solution, it would be up to the client to check that all passed options make sense...

Reply:
I hope it would be declarative, though.

Reply:
I still think this is an ok intermediate state towards a better support of attributes, but I think it would be a bad idea to release 8.8 if such checks are not done at all, since they used to be done.

Reply:
I was of the opinion of @ppedrot here, this PR didn't intend to solve the "attribute" problem, but the state problem, so it is up to the command to implement a warning.

I didn't want to put my hands into that as this is an area I thought would overlap with the attributes work. On the other hand I think there is a `command_supports_locality` function that should emit the warning, right? Plugins cannot register with it thou, but again I thought that people working on the attributes would handle that in their own PR.

Reply:
Regarding @maximedenes example, I think that if a command does not support locality, the parsing rule should not be indeed declared in the first place, so I am not sure that we can attribute the bad behavior in your example to the vernac interpretation infrastructure.

On a more general note, I think that indeed a possibility for attributes is for them to be reflect in the type of the generated rule. IMHO, dealing with this at the macro level should be better than using `Obj` hacks style. For example:

```ocaml
[local, pure] VERNAC EXTEND COMMAND foo ..
 { $e$ }
```
should require `e` to have type `?locality -> ~st -> args -> unit` etc... [using row types could also work.

Reply:
@maximedenes If you think that some action needs to be taken on this subject before the 8.8 release (and even if the attributes are on the roadmap), please open a new issue with the 8.8 milestone and a blocker label. Otherwise there is always a risk to forget.

Reply:
Looking at this PR: has something be done so that we don't release a version which stops silently dropping a locality flag, as in @maximedenes' example:
```coq
Require Import FunInd.
Local Function f (n : nat) := n.
```
Maybe should I open an issue?

Reply:
Already done, see https://github.com/coq/coq/issues/6239

Reply:
Oups :(